contributors/devel/conformance-guidelines.md

```diff
@@ -0,0 +1,76 @@
+# Conformance Test Guidelines
+
+The Kubernetes conformance test suite is a set of testcases, currently a
+subset of the integration/e2e tests, that the Architecture SIG has approved
+to define the core set of interoperable features that all Kubernetes
+deployments must support.
```

Comment:
Conformance tests in the default profile should cover features that are:
* GA (not alpha or beta APIs, nor deprecated features)
* portable (not dependent on provider-specific capabilities or on the public internet)
* not an optional feature (e.g. not policy enforcement)
* non-privileged (neither root on nodes, network, nor cluster)

Reply:
Also:
* Cannot be flaky
* Cannot skip providers (offhand, I'd say there should be no Skip... directives)

Reply:
And in the Nucleus or Application layers.
https://github.com/kubernetes/community/blob/master/contributors/devel/architectural-roadmap.md

Reply:
Would suggest a sentence immediately before or after along the lines of:

“A conformance test verifies the expected functionality works as a user might encounter it in the wild, and tests should begin by covering the most important and visible aspects of the function.”  

We don’t really define them in this doc

Reply:
Added required text. Thanks @bgrant0607 @smarterclayton